lightning/src/util/config.rs

```diff
@@ -552,6 +552,21 @@ pub struct UserConfig {
 	/// [`ChannelManager::get_intercept_scid`]: crate::ln::channelmanager::ChannelManager::get_intercept_scid
 	/// [`Event::HTLCIntercepted`]: crate::events::Event::HTLCIntercepted
 	pub accept_intercept_htlcs: bool,
+	/// If this is set to true, regardless of `accept_inbound_channels`, the user needs to manually
+	/// accept inbound requests to open a new dual-funded channel and provide any UTXOs they'd like
+	/// to contribute to the funding transaction.
+	///
+	/// When set to true, [`Event::OpenChannelV2Request`] will be triggered once a request to open a
+	/// new inbound dual-funded channel is received through a [`msgs::OpenChannelV2`] message. In that
+	/// case, a [`msgs::AcceptChannelV2`] message will not be sent back to the counterparty node unless
+	/// the user explicitly chooses to accept the request.
+	///
+	/// Default value: false.
+	///
+	/// [`Event::OpenChannelV2Request`]: crate::events::Event::OpenChannelV2Request
+	/// [`msgs::OpenChannelV2`]: crate::ln::msgs::OpenChannelV2
+	/// [`msgs::AcceptChannelV2`]: crate::ln::msgs::AcceptChannelV2
+	pub contribute_to_dual_funded_channels: bool,
```

Comment:
Note you can have two level of settings, I think. One to say we accept dual funded channels and one to say we contribute any reserve UTXO we might have available in the wallet. A counterparty can be interested to open a dual-funded channels with us, just to pipeline multiple UTXOs or profit of more flexibility of the opening format. For the second setting, this is something LND does, if there is any dual-funding request, the wallet will contribute remaining UTXOs. Fucking unsafe but interesting to avoid idle liquidity.

Reply:
Yeah I think in the meeting last week as a first step we thought to only allow contributing to inbound open requests manually where the user must provide UTXOs but I do think we'll want to expand this to be automatic based on policies (although as you mentioned, it can be a bit scary and unsafe, so avoiding the hassle for now 😅)